Added OnClick and OnMouse actions for Canvas

KristofferStrube opened the pull request:
As described in issue #83

KristofferStrube committed:
Added OnClick and OnMouse actions for Canvas

KristofferStrube committed:
Working on implementing ImageData methids. Get works, but Put needs more work

KristofferStrube committed:
Got putImageData to work. Can now both read and write imageData to canvas's

KristofferStrube committed:
Revert "Added OnClick and OnMouse actions for Canvas"

This reverts commit f3b230e167569f4c79d78d1b07465ae7b8e609c4.

KristofferStrube committed:
Revert "Revert "Added OnClick and OnMouse actions for Canvas""

This reverts commit 96876419a012a8a9ed90a3be0ddbf5edf9270c7d.

KristofferStrube committed:
Revert "Got putImageData to work. Can now both read and write imageData to canvas's"

This reverts commit c5f8a7b9b7cc8a6f1860ccd3a849b75c03482708.

KristofferStrube committed:
Revert "Working on implementing ImageData methids. Get works, but Put needs more work"

This reverts commit 7588bb1b6be621bdf369191299e7554d65015ad0.

KristofferStrube commented:
I accidentally continued working with other things on the same branch. That has been reverted now. I'm new to making Pull Requests.

ummerland reviewed (changes requested):
Hi, perfect addition to the Canvas!
There are some other Events that can also be added however: OnPointerDown, OnPointerUp, OnPointerMove

Thaina commented:
Did `ondblclick` was included?

KristofferStrube committed:
Added OnPointer eventhandlers for canvas

KristofferStrube committed:
Added OnContextMenu and OnDblClick handlers

KristofferStrube commented:
Hey @ummerland I must have missed a notification at some point, sorry. But I added the handlers you described now. And added the handler that @Thaina requested as well together with the `OnContextMenu` handler.

ummerland commented:
Very nice, thanks @KristofferStrube

ummerland reviewed (approved):
Very good and practical addition, clean code

Thaina commented:
@ummerland Do we have any chance this would be updated in nuget? Or we need to build this ourselves? It seem this repo don't get any updated for a whole year

attilah commented:
Sorry for the late update, we coming to this package and get review the feature PR and also update the package to .NET 5.0, thanks for the contrib.

KristofferStrube commented:
@attilah regarding the conflict with `blazor.extensions.canvas.js` which is auto-generated. Which version should be used? Doesn't seem like it would actually need to change with the modifications I've made to the project.

KristofferStrube committed:
removed erronious change to JS

KristofferStrube committed:
merge

KristofferStrube committed:
merge

KristofferStrube committed:
Merge branch 'BlazorExtensions-master' into master

KristofferStrube committed:
merge again

KristofferStrube commented:
@tbm0115 I had some problems with my merge as you can see, but there are no conflicts now. You can just squash my commits.

neilgd commented:
Hi,

I'm wondering if this change made it into the release, or maybe it got lost somewhere?

I get an error _System.InvalidOperationException: Object of type 'Blazor.Extensions.Canvas.BECanvas' does not have a property matching the name 'onmousedown'_

KristofferStrube commented:
Hey @neilgd,
No this never got merged into the master branch.

sethfduke commented:
> Hey @neilgd, No this never got merged into the master branch.

Any chance this is going to be merged at some point? Been sitting here for quite a while.
